src/core/diags/factory_diags.cpp

```diff
@@ -573,6 +575,27 @@ Error Diags::ProcessContinuousWave(uint8_t aArgsLength, char *aArgs[], char *aOu
     return error;
 }
 
+Error Diags::ProcessStream(uint8_t aArgsLength, char *aArgs[], char *aOutput, size_t aOutputMaxLen)
+{
+    Error error = kErrorInvalidArgs;
+
+    VerifyOrExit(otPlatDiagModeGet(), error = kErrorInvalidState);
+    VerifyOrExit(aArgsLength > 0, error = kErrorInvalidArgs);
+
+    if (strcmp(aArgs[0], "start") == 0)
+    {
+        SuccessOrExit(error = otPlatDiagRadioTransmitStream(&GetInstance(), true));
```

Comment:
Do you need SuccessOrExit here?

Reply:
It is not really needed, but also I see it's been defined like this in other diag CLI implementations and that could be for future proofing if we have added more code after #if..else block. Currently there is none so we can remove it. Let me know your thoughts.

Reply:
Remove SuccessOrExit check as it is not needed.